spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -193,47 +193,441 @@ For example, calling an `@Outgoing` annotated method from user code will not pos
 
-==== Supported Bean scope
+==== Supported CDI scopes
 
+Implementations of the Reactive Messaging specification must support at least the following CDI scopes:
 
-==== Method signatures
+* `@ApplicationScoped` beans
+* `@Dependent` beans
 
-This section lists the method signatures that implementation must support.
+The following code gives an example of a bean annotated with `@ApplicationScoped`:
 
+[source,java]
+----
+@ApplicationScoped
+public class ApplicationScopeBeans {
+
+  @Outgoing("source")
+  public Publisher<Integer> source() {
+    return ReactiveStreams.of(id).buildRs();
+  }
+
+  @Incoming("source")
+  @Outgoing("output")
+  public int process(int i) {
+    return i + 1;
+  }
+
+  @Incoming("output")
+  public void sink(int v) {
+    System.out.println(v);
+  }
+}
+----
+
+Implementations can provide support for other scopes. 
+However the behavior is not defined.
+
+==== Supported method signatures
 
 The signature of message stream methods can have a number of different distinct types, offering differing levels of power and simplicity to application developers. Different shapes are supported depending on whether the method is a publisher, subscriber or processor, for example, a publishing stream supports returning MicroProfile Reactive Streams `PublisherBuilder`, but not `SubscriberBuilder`, the inverse is true for a subscribing stream.
 
+This section lists the methods signatures that must be supported by the Reactive Messaging implementation.
 Implementations must validate that the stream shape matches the `@Outgoing` and `@Incoming` annotations, if they don't, a CDI definition exception should be raised to the CDI container during initialization.
 
-===== Simple method streams
+It's important to remember that users must not call these methods directly. 
+They are invoked by the Reactive Messaging implementation following the Reactive Streams protocol.
+
+Also the method must be implemented as a non-blocking fashion.
+For blocking transformations, asynchronous variants can be used.
+
+INFO: _assembly_ time is when the Reactive Messaging implementation initializes itself and creates the different bean instances and connects them together.
+INFO: In the following lists, `Message` can be an implementation of the `Message` interface.
+
+===== Methods producing data
+
+[cols="2a,1,1",options="header"]
+|===
+|Signature
+|Behavior
+|Invocation
+
+|
+[source, java]
+----
+@Outgoing("name") 
+Publisher<Message<O>> method()
+----
+| Returns a stream of `Message` associated with the channel `name`. 
+| Method called once at assembly time.
+
+|
+[source, java]
+----
+@Outgoing("channel") 
+Publisher<O> method()
+----
+| Returns a stream of _payload_ associated with the channel `channel`. Produced payloads are mapped to `Message<O>` by the Reactive Messaging implementation.
```

Comment:
regarding "Returns a stream of _payload_ associated with the channel `channel`." 
I understand that _payload_ is useda s the name of the parameter from "Message<T> of(T payload){ return ()-> payload; }" but in the text above this maps to "O". 
In the javadoc we use Message<T> so I think the above would be better as "Message<T>" (T for Type rather than O for Object). I think we should use the string 'payload' consistently - so that if it is used for the instance of the type T in the javadoc then we should only use it for the same thing here in the spec, for as an aid for those
trying to understand what is going on.

Reply:
We cannot really use `<T>` as it would not work for `Processor` using 2 parameters `<I>` and `<O`>. 

I'm updating the sentence to say: `Returns a stream of _payload_ of type `O` associated with the channel `channel`.` Is it better?

Reply:
Yes that is better. Thanks (and apologies for the delay in reply, I have off due to family issues).